core/app/models/spree/country.rb

```diff
@@ -13,9 +13,12 @@ def self.states_required_by_country_id
     end
 
     def self.default
-      if default_country_id = Spree::Config[:default_country_id]
-        find_by_id(default_country_id)
-      end || first
+      if Spree::Config.default_country_id
+        ActiveSupport::Deprecation.warn("Setting your default country via its ID is deprecated. Please set your default country via the `default_country_iso` setting.", caller)
+        find_by(id: Spree::Config.default_country_id) || find_by!(iso: Spree::Config.default_country_iso)
```

Comment:
So the scenario of find by ID returning `nil` is essentially a misconfigured shop. Previously, in that case it would return `first`, i.e. any country. I find it more fitting to actually return something sensible, and that would be the configured country by `iso`. 

It's supposed to be nudging people into configuring their shops correctly. Apart from the address form and (newly) taxes, the `default country` is also used for stock locations, and should make sense there, too.